src/internal.c

```diff
@@ -21533,6 +21550,23 @@ int ProcessReplyEx(WOLFSSL* ssl, int allowSocketErr)
                                        &ssl->curRL, &ssl->curSize);
 
 #ifdef WOLFSSL_DTLS
+#ifdef WOLFSSL_DTLS_CID
+            if (ssl->options.dtls) {
+                if (!ssl->buffers.dtlsCtx.processingPendingRecord)
+                    ssl->buffers.dtlsCtx.processingPendingRecord = 1;
+                else {
+                    ssl->buffers.dtlsCtx.processingPendingRecord = 0;
+                    if (ssl->buffers.dtlsCtx.pendingPeer.sa != NULL) {
+                        /* Clear the pending peer */
+                        XFREE(ssl->buffers.dtlsCtx.pendingPeer.sa, ssl->heap,
+                              DYNAMIC_TYPE_SOCKADDR);
+                        ssl->buffers.dtlsCtx.pendingPeer.sa = NULL;
+                        ssl->buffers.dtlsCtx.pendingPeer.sz = 0;
+                        ssl->buffers.dtlsCtx.pendingPeer.bufSz = 0;
+                    }
+                }
+            }
+#endif
```

Comment:
this and the other block to update the peer feel misplaced here. One reason is that it will run for every packet, even the ones where we don't have set any pending peer.  Another reason is that is adding more noise in the already quite convoluted processReply. 

But I can't find a good solution because I don't see a good point where we can clear the pending peer on error, as we have many exit points in this function.

The only idea I've is to modify `wolfssl_inject` so that it also invokes `wolfSSL_negotiate`  if necessary and `wolfSSL_read`. If there are application-data we can return them from `wolfssl_inject`.

The basic idea is that we include what the app is doing inside `dispatchExistingConnection` inside the `wolfSSl_inject` API.

If this makes sens to you  we can include the the peer in the  api itself. It's true what we are doing two things in a single API, but it's also true that the data that we want to inject usually have a source, so the two information are related.

Reply:
If you reject the `inject` modification, can you move this and the other block in separate functions and adds a longer comments on why we are doing that?

Reply:
> run for every packet

Only bit flipping will run for every class.

> But I can't find a good solution because I don't see a good point where we can clear the pending peer on error, as we have many exit points in this function.

That is precisely the reason for this. If we could have a single `cleanup` goto it would make it easier but this function is complicated enough.

I don't want to add anything else to the inject API. I think it should do only what the user of an inject API would expect. I'll add more documentation and refactor it to make it cleaner.